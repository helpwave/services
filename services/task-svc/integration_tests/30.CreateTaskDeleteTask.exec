#!/bin/bash

echo "no optionals"
id=$(grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"name": "Some Great Task", "patient_id": "de70d0d0-94da-4d00-9dfc-54723320fe47", "public": false}' \
	localhost:3001 \
	services.task_svc.v1.TaskService/CreateTask | jq -r ".id")

echo "result:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d "{\"id\": \"$id\"}" \
	localhost:3001 \
	services.task_svc.v1.TaskService/GetTask | jq -e '.id = "RANDOM ID"'

echo "delete again"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d "{\"id\": \"$id\"}" \
	localhost:3001 \
	services.task_svc.v1.TaskService/DeleteTask
echo "see:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d "{\"id\": \"$id\"}" \
	localhost:3001 \
	services.task_svc.v1.TaskService/GetTask 2>&1 || echo ""

echo "with optionals"
id=$(grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"name": "Some Great Task", "patient_id": "de70d0d0-94da-4d00-9dfc-54723320fe47", "public": true, "description": "This is a description.", "initial_status": "TASK_STATUS_IN_PROGRESS", "due_at": "2024-01-01T16:52:45.762Z"}' \
	localhost:3001 \
	services.task_svc.v1.TaskService/CreateTask | jq -r ".id")

echo "result:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d "{\"id\": \"$id\"}" \
	localhost:3001 \
	services.task_svc.v1.TaskService/GetTask | jq -e '.id = "RANDOM ID"'

echo "delete again"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d "{\"id\": \"$id\"}" \
	localhost:3001 \
	services.task_svc.v1.TaskService/DeleteTask
echo "see:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d "{\"id\": \"$id\"}" \
	localhost:3001 \
	services.task_svc.v1.TaskService/GetTask 2>&1 || echo ""

