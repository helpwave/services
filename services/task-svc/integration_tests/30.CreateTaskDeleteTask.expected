no optionals
result:
{
  "id": "RANDOM ID",
  "name": "Some Great Task",
  "status": "TASK_STATUS_TODO",
  "assignedUserId": "00000000-0000-0000-0000-000000000000",
  "patient": {
    "id": "de70d0d0-94da-4d00-9dfc-54723320fe47",
    "name": "Gabriel Crotty"
  },
  "dueAt": "1970-01-01T00:00:00Z",
  "createdBy": "18159713-5d4e-4ad5-94ad-fbb6bb147984",
  "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406"
}
delete again
{}
see:
ERROR:
  Code: InvalidArgument
  Message: id not found

with optionals
result:
{
  "id": "RANDOM ID",
  "name": "Some Great Task",
  "description": "This is a description.",
  "status": "TASK_STATUS_IN_PROGRESS",
  "assignedUserId": "00000000-0000-0000-0000-000000000000",
  "patient": {
    "id": "de70d0d0-94da-4d00-9dfc-54723320fe47",
    "name": "Gabriel Crotty"
  },
  "public": true,
  "dueAt": "2024-01-01T16:52:45.762Z",
  "createdBy": "18159713-5d4e-4ad5-94ad-fbb6bb147984",
  "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406"
}
delete again
{}
see:
ERROR:
  Code: InvalidArgument
  Message: id not found

